src/main/java/io/debezium/connector/yugabytedb/YugabyteDBChangeRecordEmitter.java

```diff
@@ -272,24 +272,27 @@ private Optional<DataCollectionSchema> newTable(TableId tableId) {
         LOGGER.debug("Creating a new schema entry for table: {} and tablet {}", tableId, tabletId);
         refreshTableFromDatabase(tableId);
         final TableSchema tableSchema = schema.schemaForTablet(tableId, tabletId);
-
         if (tableSchema == null) {
-            LOGGER.warn("cannot load schema for table '{}'", tableId);
             return Optional.empty();
         }
         else {
-            LOGGER.debug("refreshed DB schema to include table '{}'", tableId);
             return Optional.of(tableSchema);
         }
     }
 
     private void refreshTableFromDatabase(TableId tableId) {
         try {
             // Using another implementation of refresh() to take into picture the schema information too.
-            LOGGER.debug("Refreshing schema for the table {}", tableId);
-            schema.refresh(connection, tableId,
-                           connectorConfig.skipRefreshSchemaOnMissingToastableData(),
-                           schema.getSchemaPBForTablet(tableId, tabletId), tabletId);
+            if (connectorConfig.qlType().equals("ysql")) {
+                schema.refresh(connection, tableId,
+                        connectorConfig.skipRefreshSchemaOnMissingToastableData(),
+                        schema.getSchemaPBForTablet(tableId, tabletId), tabletId);
+
+            } else {
+                schema.refresh(tableId, connectorConfig.skipRefreshSchemaOnMissingToastableData(),
```

Comment:
Put a comment before this explaining why are we not passing the `connection` object here or why is connection not required in this case.